pick/__init__.py

```diff
@@ -31,14 +31,21 @@ def __init__(self, options, title=None, indicator='*', default_index=0, multisel
         self.indicator = indicator
         self.multiselect = multiselect or multi_select
         self.min_selection_count = min_selection_count
+        self.max_selection_count = max_selection_count
         self.options_map_func = options_map_func
-        self.all_selected = []
+
+        if multiselect is False and pre_select_all is True:
+            raise ValueError('pre_select_all only works for multiple selections')
+        self.all_selected = list(range(len(options))) if pre_select_all else []
```

Comment:
`pre_select_all` doesn't seem to work well with `min_selection_count` and `max_selection_count`

Reply:
I did not mean to add this in the same commit, so I will remove it.